system/types/app_settings.go

```diff
@@ -177,6 +181,12 @@ type (
 	}
 )
 
+func (set *ExternalAuthProvider) ValidConfiguration() bool {
+	return (set.Enabled && set.Handle != "" && set.Key != "" && set.Secret != "" &&
```

Comment:
break this into a series of `if ... { false }` checks for readability please.

Reply:
Done